test/lit/passes/remove-unused-brs.wast

```diff
@@ -26,4 +26,261 @@
+  ;; CHECK-NEXT:   (i32.const 300)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-value-effectful-corner-case-1 (param $x i32) (result i32)
+    (block $x (result i32)
+      (drop
+        (br_if $x
+          (block (result i32)
+            (call $nothing)
+            (i32.const 100)
+          )
+          (block (result i32)
+            (call $nothing)
+            (local.get $x)
+          )
+        )
+      )
+      ;; the condition cannot be reordered with this
+      (call $nothing)
+      (i32.const 300)
+    )
+  )
+
+  ;; CHECK:      (func $get-i32 (result i32)
+  ;; CHECK-NEXT:  (i32.const 400)
+  ;; CHECK-NEXT: )
+  (func $get-i32 (result i32)
+    (i32.const 400)
+  )
+
+  ;; CHECK:      (func $restructure-br_if-value-effectful-corner-case-2 (param $x i32) (result i32)
+  ;; CHECK-NEXT:  (block $x (result i32)
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (br_if $x
+  ;; CHECK-NEXT:     (block $block (result i32)
+  ;; CHECK-NEXT:      (call $nothing)
+  ;; CHECK-NEXT:      (i32.const 100)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:     (block $block2 (result i32)
+  ;; CHECK-NEXT:      (call $nothing)
+  ;; CHECK-NEXT:      (local.get $x)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (i32.const 300)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (call $get-i32)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-value-effectful-corner-case-2 (param $x i32) (result i32)
+    (block $x (result i32)
+      (drop
+        (br_if $x
+          (block (result i32)
+            (call $nothing)
+            (i32.const 100)
+          )
+          (block (result i32)
+            (call $nothing)
+            (local.get $x)
+          )
+        )
+      )
+      (drop (i32.const 300))
+      ;; the condition cannot be reordered with this
+      (call $get-i32)
+    )
+  )
+  ;; CHECK:      (func $restructure-br_if-value-effectful-corner-case-3 (param $x i32) (result i32)
+  ;; CHECK-NEXT:  (block $x (result i32)
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (br_if $x
+  ;; CHECK-NEXT:     (block $block (result i32)
+  ;; CHECK-NEXT:      (call $nothing)
+  ;; CHECK-NEXT:      (i32.const 100)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (call $nothing)
+  ;; CHECK-NEXT:   (i32.const 100)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-value-effectful-corner-case-3 (param $x i32) (result i32)
+    (block $x (result i32)
+      (drop
+        (br_if $x
+          ;; we can't do an if because of effects here
+          (block (result i32)
+            (call $nothing)
+            (i32.const 100)
+          )
+          (local.get $x)
+        )
+      )
+      ;; and we can't do a select because of effects here
+      (call $nothing)
+      ;; the condition cannot be reordered with this
+      (i32.const 100)
```

Comment:
Isn't the instruction the condition cannot be reordered with not this `(i32.const 100)` but `(call $nothing)`?

Reply:
Right, thanks, this was a copy-paste that I forgot to remove, will fix...